ramp4-pcar4/storylines-editor
# Implement minor slide editor redesign, improve 'full slide' functionality

### Related Item(s)
Issues:
#450
#453 
#457 

### Changes
- Implements a minor redesign of the slide editor.
  - Adjusts the header/common section to enhance clarity.
    - [#453] Adds more spacing between slide options to clarify which checkbox belongs to which. 
  - Round-ifys buttons and inputs so they match the ToC and metadata editor.
  - Unifies the outlines to a common 'royal blue' colour upon keyboard focus, to make it more clear what's been selected.
  - Moves most input labels to be a line before the input, which prevents overlap on small screens and matches design on metadata editor.
  - Adds a very minor shadow to inputs to emphasize their existence.
  - Updates the warning modals for `Change content type` and `Make the current panel the full slide` to use the new, clearer design.
  - Adjusts the styles of the text and image editors to improve clarity. (NOTE: remaining editors haven't really been modified; could be an issue for the future).
    - Changes breakpoints for the image previews so they don't get *too* small to see (largest screen allows 3 images in a row, medium screens allow 2, mobile-like screens allow 1). 
- [#450] Changes the `Make right panel the full slide` option to `Make the current panel the full slide`.
  - Upon disabling the option, user will now be asked which side they want the new panel to be on (left or right) (See image below).
- [#457] Prevents vertical scrollbar from appearing when editor page first loaded and no slide is selected.
 

### Notes
**New common slide editor header**:
<img width="1472" alt="image" src="https://github.com/user-attachments/assets/73bd8d18-13cc-457e-9239-eed18c1c5076">

**Enabling `Make the current panel the full slide` modal**:
<img width="536" alt="image" src="https://github.com/user-attachments/assets/f5d55504-54d2-4bc2-916e-a6f826754e39">

**Disabling `Make the current panel the full slide` modal**:
<img width="503" alt="image" src="https://github.com/user-attachments/assets/389ec0ca-7f8f-4bab-b356-3cab23370085">

**Change panel type modal**:
<img width="438" alt="image" src="https://github.com/user-attachments/assets/49f5bab8-e3f5-4059-811b-d13a44cd250b">

**Image editor (large screens)**:
<img width="1470" alt="image" src="https://github.com/user-attachments/assets/43376f99-d5ce-4cd8-aac0-840c946dcd16">

**Image editor (medium screens)**:
<img width="932" alt="image" src="https://github.com/user-attachments/assets/93b7ec0a-e97f-4812-bc6c-66558492a61e">

**Image editor (small screens)**:
<img width="344" alt="image" src="https://github.com/user-attachments/assets/75d4b43c-a0ec-4dca-99ca-64028f1b3be8">



### Testing
Steps:
1. Open any product (e.g. `00000000-0000-0000-0000-000000000000`). Immediately notice there's no vertical scroll bar for the slide editor area.
2. Open any slide. See the modified common slide elements (title, slide options, etc.) at the top.
3. Test out the `Make current panel the full slide` option. Select it, select `continue for the warning`, and the panel will be the whole slide. De-select the option, and you can choose which side to put the new panel on.
